modules/nix/default.nix

```diff
@@ -836,18 +838,18 @@ in
     users.users = mkIf configureBuildUsers nixbldUsers;
 
     # Not in NixOS module
-    users.groups.nixbld = mkIf configureBuildUsers {
+    users.groups.nixbld = {
       description = "Nix build group for nix-daemon";
       gid = config.ids.gids.nixbld;
       members = attrNames nixbldUsers;
     };
     users.knownUsers =
       let nixbldUserNames = attrNames nixbldUsers;
       in
-      mkMerge [
+      mkIf configureBuildUsers (mkMerge [
         nixbldUserNames
         (map (removePrefix "_") nixbldUserNames) # delete old style nixbld users
-      ];
+      ]);
```

Comment:
I think these should be in `users.knownUsers` unconditionally, because it means we can delete them if you start using `auto-allocate-uids`, rather than having their existence be dependent on the untracked state of whether you previously had them or not. Though I guess you can override `auto-allocate-uids` temporarily, in which case maybe we should be creating them even if it’s set…? But that would rather defeat the point…

Reply:
If we managed all `_nixbld*` users regardless of `nix.nrBuildUsers`, then I think the best solution would be to default `nix.nrBuildUsers` to 0 if `auto-allocate-uids` is on, and be okay with that iff `auto-allocate-uids` is on. Unfortunately in the current case that’s equivalent to not managing them; we should probably cap `nrBuildUsers` and use that constant for `knownUsers`, or else rework user management so that we don’t need a fixed list of every managed user. (Not expecting you to do this to fix this; just thinking out loud.)

Reply:
Ah, I see what you mean, I had a look at how NixOS does it, and it's exactly what it does. So I'll try to update it to match the NixOS behaviour.

https://github.com/NixOS/nixpkgs/blob/6d6df8288d2419766df3c8c489cf6ab60bd80e07/nixos/modules/services/system/nix-daemon.nix#L275-L280